libraries/lib-snapping/SnapUtils.cpp

```diff
@@ -246,142 +250,100 @@ namespace
-      SnapFunction("bar_1_32", XO("1/32"), SnapToBeat(32)),
-      SnapFunction("bar_1_64", XO("1/64"), SnapToBeat(64)),
-      SnapFunction("bar_1_128", XO("1/128"), SnapToBeat(128)))
-};
+      if (result < 0.0)
+         return { 0.0, false };
 
-SnapRegistryItemRegistrator triplets {
-   Registry::Placement { {},
-                         { Registry::OrderingHint::After, "beats" } },
-   SnapFunctionGroup(
-      /* i18n-hint: The music theory "triplet"*/
-      "triplets", XO("Triplets"), true,
-      SnapFunction("triplet_1_2", XO("1/2 (triplets)"), SnapToTriplets(2)),
-      SnapFunction("triplet_1_4", XO("1/4 (triplets)"), SnapToTriplets(4)),
-      SnapFunction("triplet_1_8", XO("1/8 (triplets)"), SnapToTriplets(8)),
-      SnapFunction("triplet_1_16", XO("1/16 (triplets)"), SnapToTriplets(16)),
-      SnapFunction("triplet_1_32", XO("1/32 (triplets)"), SnapToTriplets(32)),
-      SnapFunction("triplet_1_64", XO("1/64 (triplets)"), SnapToTriplets(64)),
-      SnapFunction(
-         "triplet_1_128", XO("1/128 (triplets)"), SnapToTriplets(128)))
-};
+      return SnapWithMultiplier(result, multiplier, true);
+   }
 
-SnapRegistryItemRegistrator secondsAndSamples {
-   Registry::Placement { {},
-                         { Registry::OrderingHint::After, "triplets" } },
-   SnapFunctionGroup(
-      "time", XO("Seconds && samples"), false,
-      SnapFunction("seconds", XO("Seconds"), SnapToTime(1.0)),
-      SnapFunction("deciseconds", XO("Deciseconds"), SnapToTime(10.0)),
-      SnapFunction("centiseconds", XO("Centiseconds"), SnapToTime(100.0)),
-      SnapFunction("milliseconds", XO("Milliseconds"), SnapToTime(1000.0)),
-      SnapFunction("samples", XO("Samples"), SnapToSamples))
+private:
+   const MultiplierFunctor mMultiplierFunctor;
 };
 
-SnapRegistryItemRegistrator videoFrames {
-   Registry::Placement { {},
-                         { Registry::OrderingHint::After, "time" } },
-   SnapFunctionGroup(
-      "video", XO("Video frames"), false,
-      SnapFunction(
-         "film_24_fps", XO("Film frames (24 fps)"), SnapToFrames(24.0)),
-      SnapFunction(
-         "ntsc_29.97_fps", XO("NTSC frames (29.97 fps)"),
-         SnapToFrames(30.0 / 1.001)),
-      SnapFunction(
-         "ntsc_30_fps", XO("NTSC frames (30 fps)"),
-         SnapToFrames(30.0 / 1.001)),
-      SnapFunction("film_25_fps", XO("PAL frames (25 fps)"), SnapToFrames(25.0)))
-};
+}
 
-SnapRegistryItemRegistrator cdFrames {
-   Registry::Placement { {},
-                         { Registry::OrderingHint::After, "video" } },
-   SnapFunctionGroup(
-      "cd", XO("CD frames"), false,
-      SnapFunction("cd_75_fps", XO("CDDA frames (75 fps)"), SnapToFrames(75.0)))
-};
+Registry::BaseItemPtr TimeInvariantSnapFunction(
+   const Identifier& functionId, const TranslatableString& label,
+   MultiplierFunctor functor)
+{
+   return std::make_unique<ProjectDependentMultiplierSnapItem>(
+      functionId, label, std::move(functor));
+}
```

Comment:
My latest PR "namespace-Callable" will conflict, but once this is merged into master, I will rebase, and this bit of new annoyance code will also simplify